osbs/api.py

```diff
@@ -625,8 +626,13 @@ def _do_create_prod_build(self, git_uri, git_ref,
                 # fix for module requires will have to be determined from experience.
                 raise ValueError("Flatpak build cannot be isolated")
 
-        repo_info = utils.get_repo_info(git_uri, git_ref, git_branch=git_branch,
-                                        depth=git_commit_depth)
+        try:
+            repo_info = utils.get_repo_info(git_uri, git_ref, git_branch=git_branch,
+                                            depth=git_commit_depth)
+        except yaml.scanner.ScannerError as e:
```

Comment:
This doesn't seem right to me. This context should have no information about implementation detail, that we use yaml parser.

IMHO we should update `OSBSException` to use message from `str()` instead of `repr()` in constructor

Reply:
If it's better to use str to get the error, we should do this inside get_repo_info.

Reply:
So, any opinions on changing `OSBSException` to use `str()` vs moving the error handling to `get_repo_info()`?

Reply:
IMO, I'd go with OSBSException as our goal is to always use str() instead of repr() in future.

But on the other hand, exception context should be isolated, so if we know that somewhere in `get_rep_info` is expected to get Yaml exception, we should catch it and raise as OSBS exception

Reply:
So I'm guessing I should do both?

Also, the actual line where the error happens is [in osbs.repo_utils.RepoConfiguration.\_\_init\_\_()](https://github.com/projectatomic/osbs-client/blob/f2e2ae3029bc368e4b6c1c07b420b39b24b5d61f/osbs/repo_utils.py#L64), so maybe the the try/except should be there?

Reply:
Yes, that's sounds about right